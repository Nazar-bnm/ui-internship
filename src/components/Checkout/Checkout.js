import React from 'react';
import cx from 'classnames';

import Accordion from '../Accordion';
import LoginRegister from './LoginRegister';
import BillingInfo from './BillingInfo';
// import ShippingInfo from './ShippingInfo';
// import ShippingMethod from './ShippingMethod';
// import Payment from './Payment';
// import OrderReview from './OrderReview';

import './Checkout.scss';

const CN = 'checkout';
const data = [
  {
    title: '01. Checkout',
    id: 1,
    description: <LoginRegister />
  },
  {
    title: '02. Billing Info',
    id: 2,
    description: <BillingInfo />
  }
  // {
  //   title: '03. Shipping Info',
  //   id: 3,
  //   description: <ShippingInfo />
  // },
  // {
  //   title: '04. Shipping Method',
  //   id: 4,
  //   description: <ShippingMethod />
  // },
  // {
  //   title: '05. Payment',
  //   id: 5,
  //   description: <Payment />
  // },
  // {
  //   title: '06. Order review',
  //   id: 6,
  //   description: <OrderReview />
  // }
];
// console.log('data ', data);
// console.log(data)

const Checkout = () => (
  <div className={`${cx(CN)}-wrapper content`}>
    <Accordion
      className={`${CN}__accordion`}
      heightItem="auto"
      data={data}
      open
    />
  </div>
);

export default Checkout;
