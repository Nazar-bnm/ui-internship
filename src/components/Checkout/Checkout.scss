@import '../../scss/global_styles';
@import '../Accordion/AccordionItem/AccordionItem.scss';

.checkout {
  &-wrapper {
    font-size: 0.8rem;
    font-family: font(primary-light);
  }

  &__accordion {
    .accordion-item {
      border: 1px solid $silver-chalice;
      border-top: none;

      &__header {
        padding: 20px;
        margin-top: 10px;
        border-bottom: 1px solid $silver-chalice;
        font-family: font(primary-light);
  
        &:hover,
        &:active,
        &--active {
          color: $black;
          margin-top: 10px;
        }
      }
    }
    
    .caret {
      display: none;
    }
  }
}
