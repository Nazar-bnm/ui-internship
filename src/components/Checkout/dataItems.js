export const accordionItemsData = [
  {
    id: 1,
    componentName: 'loginRegister',
    title: '01. checkout',
    description: 'Architecto, earum temporibus quidem ex eaque unde harum sit'
  },
  {
    id: 2,
    componentName: 'billingInfo',
    title: '02. billing info',
    description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'
  },
  {
    id: 3,
    componentName: 'loginRegister',
    title: 'shipping info',
    description: 'Earum temporibus quidem ex eaque unde harum sit'
  }
];
