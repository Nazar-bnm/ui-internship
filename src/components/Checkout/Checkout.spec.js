import React from 'react';
import { shallow } from 'enzyme';

import Checkout from './Checkout';

describe('<Checkout />', () => {
  let wrapper = shallow(<Checkout />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
