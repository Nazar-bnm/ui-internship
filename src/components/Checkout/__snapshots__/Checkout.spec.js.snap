// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> should match snapshot 1`] = `
<div
  className="checkout__wrapper content"
>
  <Accordion
    className="checkout__accordion"
    data={
      Array [
        Object {
          "description": <Memo(Connect(LoginRegister)) />,
          "id": 1,
          "title": "01. Checkout",
        },
        Object {
          "description": <Memo(Connect(BillingInfo)) />,
          "id": 2,
          "title": "02. Billing Info",
        },
        Object {
          "description": <ShippingInfo />,
          "id": 3,
          "title": "03. Shipping Info",
        },
        Object {
          "description": <ShippingMethod />,
          "id": 4,
          "title": "04. Shipping Method",
        },
        Object {
          "description": <Memo(Connect(Payment)) />,
          "id": 5,
          "title": "05. Payment",
        },
        Object {
          "description": <OrderReview />,
          "id": 6,
          "title": "06. Order review",
        },
      ]
    }
    heightItem="auto"
    labels={
      Array [
        "Required*",
        "Required*",
        "",
        "",
        "Required*",
        "",
      ]
    }
    open={false}
    scroll={false}
    showAll={false}
  />
</div>
`;
